src/index.ts

```diff
@@ -34,25 +34,24 @@ export async function resolve(
   context: ResolveContext,
   defaultResolve: Function,
 ): Promise<ResolveResponse> {
-  let { parentURL = baseURL } = context;
+  const { parentURL = baseURL } = context;
 
-  let forceRelative = false;
+  // let forceRelative = false;
   if (TSConfig?.paths) {
     for (const tsPath of Object.keys(TSConfig.paths)) {
-      const tsPathKey = tsPath.replace('/*', '');
+      const tsPathKey = tsPath.replace('*', '');
       if (specifier.startsWith(tsPathKey)) {
-        const pathSpecifier = TSConfig.paths[tsPath][0].replace(
-          '/*',
-          specifier.split(tsPathKey)[1],
+        const tsResolvedPath = resolvePath(
+          resolvePath(baseURL, TSConfig.baseUrl!),
+          TSConfig.paths[tsPath][0].replace('*', specifier.split(tsPathKey)[1]),
         );
 
-        forceRelative = true;
+        specifier = `./${relative(
+          dirname(fileURLToPath(parentURL)),
+          tsResolvedPath,
+        )}`;
```

Comment:
Annoyingly I can't find a way to get relative to not return `Utils/Math` when I want it to be `./Utils/Math` in the example tests.

Reply:
So even though this would/could/does return stuff like `./../../Utils/Math` the `./../OTHER_PATH` is a proper path.